### PR TITLE
Deprecate jaeger exporters.

### PR DESCRIPTION
**Pull request by breedx-splk**
Earlier this year, jaeger acquired [native support for trace ingest via OTLP](https://www.jaegertracing.io/docs/1.38/apis/#opentelemetry-protocol-stable). Additionally, the language-specific client libraries that contained the jaeger thrift exporter implementations were [EOLed via archival](https://github.com/jaegertracing?q=client&type=all&language=&sort=). At the top of the READMEs for those archived clients, [Jaeger recommends switching to OpenTelemetry](https://github.com/jaegertracing/jaeger-client-java#-this-library-is-deprecated).

Since the client repos were archived, they will not receive security updates. Otel languages whose jaeger exporters depend on those clients will therefore need to reimplement the exporters or drop support for them. Because jaeger is actively urging users to switch to OpenTelemetry clients, I think that deprecation makes sense. 

Continued support for jaeger exporters is a maintenance burden for otel language maintainers. Furthermore, the existence of so many exporter choices often serves to confuse users (eg. "I'm running jaeger and there's an otel jaeger exporter, so I should use that one?") and makes understanding otel <-> jaeger interoperation more difficult.

Related issue: https://github.com/open-telemetry/opentelemetry-specification/issues/2859

### CODE REVIEW

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by carlosalberto** [comment]
cc @yurishkuro

File: specification/sdk-environment-variables.md [hunk 1]
```diff
@@ -157,6 +162,7 @@ The value MUST be one of:
 [jaeger_http]: https://www.jaegertracing.io/docs/latest/apis/#thrift-over-http-stable
 [jaeger_grpc]: https://www.jaegertracing.io/docs/latest/apis/#protobuf-via-grpc-stable
 [jaeger_udp]: https://www.jaegertracing.io/docs/latest/apis/#thrift-over-udp-stable
+[jaeger_otlp]: https://www.jaegertracing.io/docs/1.38/apis/#opentelemetry-protocol-stable
```
[the remark this answers is not in the document]
Comment (breedx-splk) [reply specification/sdk-environment-variables.md]: Absolutely 💯 . We can link to https://opentelemetry.io/docs/reference/specification/protocol/exporter/ instead if we want, but I wanted to keep the links consistent. Regardless, I will update `1.38` to `latest` in the url (whoops). 

If folks generally think the direct link to the OTLP exporter docs are better, I'm fine with changing that.

[the remark this answers is not in the document]
Comment (jack-berg) [reply specification/sdk-environment-variables.md]: Oh sorry - missed that this was a ref to a link used further up. I thought `jaeger_otlp` was being added as an option for the `OTEL_EXPORTER_JAEGER_PROTOCOL` variable. Please disregard!

**Comment by carlosalberto** [comment]
@yurishkuro Ping

**Comment by yurishkuro** [comment]
@carlosalberto discussion still ongoing in https://github.com/open-telemetry/opentelemetry-specification/issues/2859

**Comment by breedx-splk** [comment]
Definitely not stale. Finishing up blog post here https://github.com/open-telemetry/opentelemetry.io/pull/1892 and looking into cross posting with Jaeger blog.

File: specification/trace/sdk_exporters/jaeger.md [hunk 2]
```diff
@@ -7,10 +7,11 @@ The generic transformation [rules specified here](../../common/mapping-to-non-ot
 particular generic transformation rule and the rule in this document contradict
 then the rule in this document MUST be used.
 
-Jaeger accepts spans in two formats:
+Jaeger accepts spans in three formats:
```
Comment (reyang) [inline specification/trace/sdk_exporters/jaeger.md:10]:
```suggestion
Jaeger accepts spans in the following formats:
```

**Review by reyang — APPROVED** [review]
Approving assuming that the PR will switch to Ready for Review after the communications are done.

**Comment by breedx-splk** [comment]
> Approving assuming that the PR will switch to Ready for Review after the communications are done.

Yeah, and I think we're close...just need some additional input on deprecation period length on #2859.

**Review by Kielek — APPROVED** [review]

**Review by jack-berg — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by MrAlias — APPROVED** [review]

File: specification/sdk-environment-variables.md [hunk 3]
```diff
@@ -163,7 +163,14 @@ See [OpenTelemetry Protocol Exporter Configuration Options](./protocol/exporter.
 
 ## Jaeger Exporter
 
-**Status**: [Stable](document-status.md)
+**Status**: [Deprecated](document-status.md)
```
Comment (arminru) [inline specification/sdk-environment-variables.md:166]: Please also add an entry in CHANGELOG.md

**Review by arminru — APPROVED** [review]

**Review by carlosalberto — APPROVED** [review]

**Comment by carlosalberto** [comment]
We are almost ready to merge this one. I'd like a final review cycle by @yurishkuro and we are good to go.

**Review by yurishkuro — APPROVED** [review]

**Pull request merged**